From 8a51e45116e202ec077b53b47b0c9b9ee5a55f91 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Mon, 28 Oct 2013 17:26:08 +0800
Subject: [PATCH 14/45] ENGR00281405 Android reboot occur if do multiple
 operations at

Settings->Developer options menu. 30%

Check boundary before call mPointers.get(id)

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 core/java/com/android/internal/widget/PointerLocationView.java | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/core/java/com/android/internal/widget/PointerLocationView.java b/core/java/com/android/internal/widget/PointerLocationView.java
index e339c44..b7e5a2a 100644
--- a/core/java/com/android/internal/widget/PointerLocationView.java
+++ b/core/java/com/android/internal/widget/PointerLocationView.java
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2010 The Android Open Source Project
- *
+ * Copyright (C) 2013-2015 Freescale Semiconductor, Inc.
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
@@ -596,6 +596,14 @@ public class PointerLocationView extends View implements InputDeviceListener,
         }
         for (int i = 0; i < NI; i++) {
             final int id = event.getPointerId(i);
+
+			//add bound check
+			int size = mPointers.size();
+			if(id >= size) {
+				Log.w(TAG, "addPointerEvent, id " + id + " >= size " + size);
+				return;
+			}
+
             final PointerState ps = mCurDown ? mPointers.get(id) : null;
             final PointerCoords coords = ps != null ? ps.mCoords : mTempCoords;
             event.getPointerCoords(i, coords);
-- 
1.8.0

